### DISCUSSION TITLE
Popover Component does not close when clicking outside Popover

### DISCUSSION THREAD

**Discussion by cheestudio** [WordPress/gutenberg#45748] [Q&A]
### Description

When a Popover is opened, if the user clicks outside the Popover's contents, it should close without having to re-trigger the open/close function. The "context" menu (three vertical dots) on BlockControls works this way already, which is the expected behavior.

### Step-by-step reproduction instructions

1. Create a block using create-block package
2. Implement a Popover 
3. Click to open the Popover
4. Click anywhere else, the Popover will remain open

### Screenshots, screen recording, code snippet

![chrome_5bKkBF7OWN](https://user-images.githubusercontent.com/2074095/201528264-c91b2f0b-19dc-46f2-86b6-b6837f155132.gif)

### Environment info

WordPress 6.1, custom theme
Chrome 107

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

**Comment by t-hamano** [comment]
Hi @cheestudio,

I think that nothing should happen when you click outside the popover unless you explicitly close it by `onClose`.

A basic implementation would be as follows:

```javascript
const [ isOpen, setIsOpen ] = useState( false );

<Button onClick={() => setIsOpen( true ) }>Toggle Popover</Button>

{ isOpen && (
	<Popover
		onClose={() => setIsOpen( false ) }
	>
		Popover Content
	</Popover>
) }
```

**Comment by cheestudio** [comment]
I see, I was not aware I had to explicitly set define a function for this. Using your example works great and does indeed clear the Popover when you click outside of the modal, but now if you click the Button again once the Popover is open, the Popover closes and re-opens instantly. Do you have an example of the two actions working in tandem? This is how I have my state handler set up:

```js
const [iconIsVisible, iconPopupVisible] = useState(false);

const toggleIconPopup = () => {
    iconPopupVisible((prevState) => !prevState);
}

<Button onClick={toggleIconPopup} name="iconPopup">
   Open Icon Popover
</Button>

{ iconIsVisible &&
        <Popover
            onClose={() => iconPopupVisible(false)}
        >
            Popover Content
        </Popover>
}

**Comment by t-hamano** [comment]
Popover may need to be included in the button component.
https://developer.wordpress.org/block-editor/reference-guides/components/popover/#usage

Then, since this is a question, not an issue, I'll convert it to discussion.
